### PR TITLE
Decouple unit tests and duckdb from compilation.

### PR DESCRIPTION
**Pull request by pedroerp**
Summary: Adding BUILD_TESTING and VELOX_ENABLE_DUCKDB to control whether
to compile unit tests and duckDB code. BUILD_TESTING requires duckDB to
be compiled for differential testing, so cmake will fail in case duckDB
is disabled.

### CODE REVIEW

**Review by aditi-pandit — APPROVED** [review]
This really helps for a more efficient build. Thanks for these changes !

**Review by wenleix — APPROVED** [review]
Thanks!

File: Makefile [hunk 1]
```diff
@@ -43,8 +54,13 @@ clean:					#: Delete all build artifacts
 	rm -rf $(BUILD_BASE_DIR)
 
 cmake:					#: Use CMake to create a Makefile build system
-	mkdir -p $(BUILD_BASE_DIR)/$(BUILD_DIR) && \
-	cmake -B "$(BUILD_BASE_DIR)/$(BUILD_DIR)" $(GENERATOR) $(USE_CCACHE) $(FORCE_COLOR) ${WARNINGS_AS_ERRORS} ${ENABLE_ALL_WARNINGS} -DCMAKE_BUILD_TYPE=$(BUILD_TYPE)
+	mkdir -p $(BUILD_BASE_DIR)/$(BUILD_DIR)
```
Comment (funrollloops) [inline Makefile:57]: the mkdir is actually redundant; cmake will create the binary directory automatically.

**Review by funrollloops — COMMENTED** [review]
@mshang816 has a conflicting PR:
https://github.com/facebookincubator/velox/pull/377

Which one do we want to go forward with?

**Comment by pedroerp** [comment]
> @mshang816 has a conflicting PR: #377
> 
> Which one do we want to go forward with?

Let's go forward with #377, but please add the stuff on this PR first. Just made some comments there as well.

**Comment by mbasmanova** [comment]
Closing stale PRs. Feel free to re-open if decide to continue working on this.

**Pull request closed without merging**